### DISCUSSION TITLE
Quo vadis? - The way ahead from a (end)-users perspective...

### DISCUSSION THREAD

**Discussion by pbaeumel** [probonopd/irdb#97] [General]
Hi @probonopd,
Hi @bengtmartensson,

I would categorize myself as an "somehow experience (end-)user". Having a Logitech Harmony Elite controlling a Samsung TV, HDMI Switch, NVIDIA Shield Pro and a linux-based settop-box "Reelbox Avantgarde" (based on [VDR](http://tvdr.de) [Video Disk Recorder by Klaus Schmidinger]). Somewhen in the feature I will Update to an Intel NUC also containing VDR.

I`m currently worried about Logitech discontiuning the Harmonys and there the uncertancy of how long they will "hold". Mainly because they are in respect to the IR-Codes used "closed source" in my eyes... (For the Reelbox-Devices in possesion I must admit that I didn`t keep any original remote control anymore as they have a rubbery finish at the back which tend to go gooey; so alone here the only fallback option is the closed harmony...)

By surfing around last week between the YIO Remote, VDR, Android/iOS-Softwarebased Remotes, etc. I only found out about the exsistence of the Global Cache Control Tower Database and your good database as well.

As one can read and absolutely understand from:
#65, #58 as well as #77 (I confess; I still havent got it!) the project stands on crossroads? Am I correct?
Personally I would love to share the codes for my old Reelbox Devices to your project; but this is obviously somewhat difficzlt as @bengtmartensson already commented. From a format viewpoint I would also prefer the XML-Format know as Girr.

But something catches my mind also which has not been discussed yet: What about esential metadata like the manufacturer and the model of the controlled device! Of course I understand that this is database of remotes! But in reality an (End-)user will more likely search the files according to the manufacturer of the controlled device; for example I would always serach for:
Manufacturer: "Samsung"; Model: "GU43TU6979" for my TV rather then glimpsing that I should search for the term "BN59-01315B" (which is the model code of the remote itself; not to mention any manufacturer who would not print a model-no on the remote itself).

And this metadata of the "controlled device" with have at least a 1:n (on some occassions an n:n-connection); being a remote control used by a great number of controlled devices. I strongly would suggest to consider this metadata.

Coming back to my original question:
You have already talked about a new data-format, that you would prefer static and free hosting.
@bengtmartensson already mentioned a SQLite-Database as an example/possibility.
Is there some way to help you guys out?! I consider this database as valuable and useful and would love to support you guys by keeping it alive. Just tell us how we could help out.

(At the moment I`m also looking into an old but exsisting licence of [Woltlabs](http://www.woltlab.com) Products; for example their Filebase may provide an "easy" way getting to a website. Of course that wouldn`t be static; but maybe some future day might be sponsored. Of course the Outputs of such an "Interface" could/should somehow flow back to Github.

Best regards,
Patrick

**Comment by bengtmartensson** [comment]
The shorcomings of the current irdb has been discussed many times. I do not think any major changes are planned or to be expected. Take it for what it is, use it for what it can be used for, but do not expect it to develop into something radically different. I practices what I preach, and wrote RemoteLocator #63, which can be used from within IrScrutinizer. It (IMHO) eliminates the need of the irdb.tk website. 

Patrick, if you desire, you can of course submit your Girr files to [GirrLib](https://github.com/bengtmartensson/GirrLib), assuming that they hold a certain quality (for example, metadata :-)).

Of course,writing for example am SQL data base as a back end for Girr would be a cool project. Question is, would it be useful too, attract users etc. As it stands, a file system with Girr (or CVS) files, organized as _manufaturer/deviceclass/model_ together with xpath for seaching within those files can be pretty useful too. The only real advantage is that of a relational data base; you can do all sort of funny searches that you cannot do with grep + xpath. Performance is think is a non-issue for all relevant use cases.

**Comment by probonopd** [comment]
This repository is meant for developers more than for end users. End user facing projects can use the codes contained in this repository. For example, tools like IrScrutinizer can make use of these codes.
